### PR TITLE
Adding parsing of /proc/<pid>/status

### PR DESCRIPTION
**Pull request by samber**
@grobie

Adding some useful data, such as context switches count, full memory usage...

### CODE REVIEW

File: proc_status.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		kv := strings.SplitN(line, ":", 2)
+
+		// removes spaces
+		k := string(strings.TrimSpace(kv[0]))
+		v := string(strings.TrimSpace(kv[1]))
+		// removes "kB"
+		v = string(bytes.Trim([]byte(v), " kB"))
+
+		// value to int when possible
+		vKBytes, _ := strconv.ParseUint(v, 10, 64)
```
Comment (discordianfish) [inline proc_status.go]: Check error

File: proc_status.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+		// convert kB to B
+		vBytes := vKBytes * 1024
+
+		if k == "Name" {
```
Comment (discordianfish) [inline proc_status.go]: Use switch instead

File: proc_status.go (hunk 1 above)
Comment (samber) [reply proc_status.go]: Hi @discordianfish 

This error check is not necessary (see comment).

Comment (samber) [reply proc_status.go]:
```
// we can skip error check here, 'cause vKBytes is not used when value is a string
```

File: proc_status.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	VmSwap uint64
+	// Size of hugetlb memory portions
+	HugetlbPages uint64
+
+	// Number of voluntary context switches.
+	VoluntaryCtxtSwitches uint64
+	// Number of involuntary context switches.
+	NonVoluntaryCtxtSwitches uint64
+
+	fs FS
```
Comment (pgier) [inline proc_status.go]: Is this `fs` necessary?  I don't see it used anywhere.

**Comment by samber** [comment]
Rebased ;)

**Comment by pgier** [comment]
Great!  Could you also squash these commits with a descriptive commit log message?  Just want to avoid commit logs like "Oops".

**Comment by samber** [comment]
@pgier done

**Comment by pgier** [comment]
LGTM

**Comment by discordianfish** [comment]
LGTM but again naming.. We should decide on #44 asap.

**Comment by pgier** [comment]
Merged.  We can fix any naming issues related to #44 in a separate PR.

**Pull request merged**